sdk/python/cmd/pulumi-language-python/main.go

```diff
@@ -210,6 +209,8 @@ func (host *pythonLanguageHost) GetRequiredPlugins(ctx context.Context,
 		return nil, err
 	}
 
+	validateVersion(ctx, host.virtualenvPath)
```

Comment:
@t0yv0 I've moved it to GetRequiredPlugins as that is where the virtualEnv is configured(if needed) an dependencies are installed. It seems to make sense to put it here, but it depends on GetRequiredPlugins being invoked by something else prior to a Run call. It sounds like an ok assumption since if we didn't call this, programs would break.